server/src/main/java/org/apache/gravitino/server/web/rest/CatalogOperations.java

```diff
@@ -187,6 +187,70 @@ public Response testConnection(
     }
   }
 
+  @GET
+  @Path("{catalog}/activate")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "activate-catalog." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "activate-catalog", absolute = true)
+  public Response activateCatalog(
+      @PathParam("metalake") String metalake, @PathParam("catalog") String catalogName) {
+    LOG.info("Received activate request for catalog: {}.{}", metalake, catalogName);
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            NameIdentifier ident = NameIdentifierUtil.ofCatalog(metalake, catalogName);
+            TreeLockUtils.doWithTreeLock(
+                NameIdentifierUtil.ofMetalake(metalake),
+                LockType.READ,
```

Comment:
Why do you use `READ` here? Does it mean that multiple threads can do the activated work simultaneously?

Reply:
changed to `WRITE` lock